sympy/matrices/dense.py

```diff
@@ -305,7 +305,11 @@ def _new(cls, *args, copy=True, **kwargs):
             # It should be used directly without creating a copy.
             if len(args) != 3:
                 raise TypeError("'copy=False' requires a matrix be initialized as rows,cols,[list]")
-            rows, cols, flat_list = args
+            else:
+                rows, cols, flat_list = args
+                if rows < 0 or cols < 0:
+                    raise ValueError("Cannot create a {} x {} matrix. "
+                                     "Both dimensions must be positive".format(rows, cols))
```

Comment:
The point of copy=False is that we shouldn't need to do any checking.

Reply:
This was added for handling the negative cases in `eye` like `eye(-1, -4)`, which were not raising any error before.
Now with the code you suggested, it riases some other error due to `vals[::cols+1] = [cls.one]*min(rows, cols)`, as for `rows` or `cols` if negative, it will raise `ValueError: attempt to assign sequence of size 0 to extended slice of size 2`